specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,283 @@
+}
+
+// This is the ICoreWebView2Frame3 interface.
+[uuid(fe1d3718-fe8d-48ab-8594-9e3fff6755ac), object, pointer_default(unique)] 
+interface ICoreWebView2Frame3 : IUnknown {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// A frame will raise a `LaunchingExternalProtocol` event when any non-main
+  /// frames attempt to launch an external protocol. This event aligns with the 
+  /// event raised on the `CoreWebView2` interface. 
+
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2FrameLaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+}
+
+/// Receives `LaunchingExternalProtocol` events for main frame.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Receives `LaunchingExternalProtocol` events for non-main frame.
+[uuid(a898c12c-f949-474c-913b-428770cfc177), object, pointer_default(unique)] 
+interface ICoreWebView2FrameLaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Frame* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The uri of the requested protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The host may set this flag to cancel the protocol launch.  If set to
+  /// `TRUE`, the protocol will not be launched. If cancelled, the
+  /// dialog is not displayed regardless of the `Handled` property.
+
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+
+  /// Sets the `Cancel` property. The default value is `FALSE`.
+
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// The host may set this flag to `TRUE` to hide the dialog.
+  /// The protocol request will continue as normal if it is not
+  /// cancelled, although there will be no default UI shown. The default
+  /// value is `FALSE` and the default external protocol dialog is shown.
+
+  [propget] HRESULT Handled([out, retval] BOOL* handled);
+
+  /// Sets the `Handled` property. The default value is `FALSE`.
+
+  [propput] HRESULT Handled([in] BOOL handled);
+
+  /// The host may set this flag to `TRUE` to disable the always open checkbox. 
+  /// This will also revoke previous permissions given to this protocol/origin.
```

Comment:
So does the always open checkbox apply to just the protocol of the URI to be launched within just the origin of the page doing the launching? And is that value saved in the user data folder across sessions?

Reply:
Yes, the checkbox applies to just the protocol of the URI to be launched from that origin specifically.  The value is saved in the udf across sessions.